lib/routes.js

```diff
@@ -84,6 +84,9 @@ const setupRoutes = function setupRoutes() {
       };
 
       socket.progress = (data) => {
+        if (socket.readyState !== 1) {
```

Comment:
I'm good with the change. But it begs the question: should we wrap `socket.send` and inject that same logical check so all attempts to `socket.send` are covered?

Reply:
That sounds reasonable, it should prevent any other such problems. I'll look into it.